### PR TITLE
Handle google-auth-exception

### PR DESCRIPTION
**Pull request by theefer**
When detecting a google auth exception in the oauthCallback page, abort and return the error to the caller.

### CODE REVIEW

**Comment by theefer** [comment]
cc @jamesgorrie

File: src/session.js [hunk 1]
```diff
@@ -32,11 +33,15 @@ function waitForLocation(iframe, timeout) {
     var throttleDelay = 50; // ms
 
     return tryReadingIframeLocation(iframe).
-        catch(() => {
-            // Race with timeout which rejects the promise and breaks the
-            // infinite loop
-            return Promise.race([delay(throttleDelay), timeout]).
-                then(() => waitForLocation(iframe, timeout));
+        catch(error => {
+            if (error instanceof NotReadyException) {
```
Comment (jamesgorrie) [inline src/session.js:37]: :eyes: :boom: :+1:

File: src/session.js [hunk 2]
```diff
@@ -47,9 +52,17 @@ function tryReadingIframeLocation(iframe) {
             // we can sort of detect this by checking the location of the iframe
             // if the contentDocument object cannot be accessed, its due to a security error.
             // security error will occur when the frame is on a different origin
-            resolve(iframe.contentDocument.location);
+
+            // Heuristic to detect google auth error in the iframe document
+            var textContent = iframe.contentDocument.body.textContent;
```
Comment (jamesgorrie) [inline src/session.js]: `const`

Comment (jamesgorrie) [inline src/session.js]: Will this work if / when we move to another that this script isn't running on?

Comment (theefer) [inline src/session.js]: Changed to `const` (though there are `var`s elsewhere in this file, was an early ES6 adoption)

Comment (theefer) [inline src/session.js]: > Will this work if / when we move to another that this script isn't running on?

Not sure what you mean by that?

Comment (jamesgorrie) [inline src/session.js]: If the login page is on `media-auth.gutools` and this script is running on media.gutools, will we still have access to the `iframe.contentDocument`?

Comment (theefer) [inline src/session.js]: Hm, that is an excellent question. I'm not sure.

This change isn't really related to this domain change so I'm going to ignore this for now, but yeah I will need to review this when attempting to break out the auth again...

Thanks!

**Comment by jamesgorrie** [comment]
:+1:

**Pull request merged**